flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/StateAssignmentOperation.java

```diff
@@ -620,18 +758,112 @@ private static void checkStateMappingCompleteness(
 			chainOpParallelStates,
 			oldParallelism,
 			newParallelism);
+	}
+
+	static class TaskStateAssignment {
```

Comment:
1. Isn't it introduced by the two-pass refactoring commit?
2. It's a purely internal class of `StateAssignmentOperation`. Would it be easier because of side-by-side view or is there a different reason?

Reply:
1. Yes, but IIUC `TaskStateAssignment` could be added before
2. Reviewing is one reason, but also in general why some helper code shouldn't reside in a separate file? Member lookup would be easier for example

However, it's probably a matter of taste.

Reply:
I split the two pass commit into two, but they are strongly related, so I'm not sure if the should exist on their own.

Reply:
Thanks a lot! To me, it's easier to understand changes now.